qdrant-landing/content/documentation/guides/distributed_deployment.md

```diff
@@ -230,6 +235,153 @@ DELETE /cluster/peer/{peer_id}
 
 After that, Qdrant will exclude the node from the consensus, and the instance will be ready for shutdown.
 
+### User-defined sharding
+
+*Available as of v1.7*
+
+Qdrant allows you to specify the shard for each point individually. This feature is useful if you want to control the shard placement of your data, so that operations can hit only the subset of shards they actually need. In big clusters, this can significantly improve the performance of operations that do not require the whole collection to be scanned.
+
+A clear use-case for this feature is managing a multi-tenant collection, where each tenant (let it be a user or organization) is assumed to be segregated, so they can have their data stored in separate shards.
+
+To enable user-defined sharding, set `sharding_method` to `custom` during collection creation:
+
+```http
+PUT /collections/{collection_name}
+{
+    "shard_number": 1,
+    "sharding_method": "custom"
+    // ... other collection parameters
+}
+```
+
+```python
+from qdrant_client import QdrantClient
+from qdrant_client.http import models
+
+client = QdrantClient("localhost", port=6333)
+
+client.create_collection(
+    name="{collection_name}",
+    shard_number=1,
+    sharding_method=models.ShardingMethod.CUSTOM,
+    # ... other collection parameters
+)
+```
+
+```typescript
+import { QdrantClient } from "@qdrant/js-client-rest";
+
+const client = new QdrantClient({ host: "localhost", port: 6333 });
+
+client.createCollection("{collection_name}", {
+    shard_number: 1,
+    sharding_method: "custom",
+    // ... other collection parameters
+});
+```
+
+```rust
+
+use qdrant_client::{
+    client::QdrantClient,
+    qdrant::{CreateCollection, ShardingMethod},
+};
+
+let client = QdrantClient::from_url("http://localhost:6334").build()?;
+
+client
+    .create_collection(&CreateCollection {
+        collection_name: "{collection_name}".to_string(),
+        shard_number: Some(1),
+        sharding_method: Some(ShardingMethod::Custom),
+        ..Default::default()
+    })
+    .await?;
+```
+
+In this mode, the `shard_number` means the number of shards per shard key, where points will be distributed evenly. For example, if you have 10 shard keys and a collection config with these settings:
+```
+{
+    "shard_number": 1,
+    "sharding_method": "custom"
+    "replication_factor": 2
+}
+```
+
+Then you will have `1 * 10 * 2 = 20` total physical shards in the collection.
+
+To specify the shard for each point, you need to provide the `shard_key` field in the point payload:
```

Comment:
I think we better not name it '* payload' anywhere as the shard key isn't actually in payload data. It could be confusing for users.

```suggestion
To specify the shard for each point, you need to provide the `shard_key` field in the upsert request:
```

Reply:
I wondered why copilot kept putting it inside the payload! I wrote the wrong thing here